File: src/main/java/no/nav/syfo/domain/Godkjenning.java
package no.nav.syfo.domain;

import lombok.Data;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;

@Data
@Accessors(fluent = true)
public class Godkjenning {
    public long id;
    public long oppfoelgingsdialogId;

    public boolean godkjent;
    public boolean delMedNav;
    public String godkjentAvAktoerId;
    public String beskrivelse;
    public LocalDateTime godkjenningsTidspunkt;
    public Gyldighetstidspunkt gyldighetstidspunkt = new Gyldighetstidspunkt();
}

File: src/main/kotlin/no/nav/syfo/api/v2/domain/oppfolgingsplan/Godkjenning.kt
package no.nav.syfo.api.v2.domain.oppfolgingsplan

import java.time.LocalDateTime

data class Godkjenning(
    val godkjent: Boolean,
    val godkjentAv: Person,
    val beskrivelse: String?,
    val godkjenningsTidspunkt: LocalDateTime,
    val gyldighetstidspunkt: Gyldighetstidspunkt,
    val delMedNav: Boolean
)
